### ISSUE TITLE
generate a heat map

### ISSUE THREAD

**Issue by jun614** [luow23/INP-Former#12]
Can you provide the code to generate a heat map？Thank you！

**Comment by luow23** [comment]
Thank you for your interest. We provide a function for saving anomaly maps, which you can use for visualization. This function can be integrated into the evaluation_batch function in utils.py for use.
def save_img(imgs, anomaly_map, gt, save_root, img_path):
batch_num = imgs.shape[0]
for i in range(batch_num):
img_path_list = img_path[i].split('\')
class_name, category, idx_name = img_path_list[-4], img_path_list[-2], img_path_list[-1]
os.makedirs(os.path.join(save_root, class_name, category), exist_ok=True)
input_frame = denormalize(imgs[i].clone().squeeze(0).cpu().detach().numpy())
cv2_input = np.array(input_frame, dtype=np.uint8)
plt.imsave(os.path.join(save_root, class_name, category, fr'{idx_name}_0.png'), cv2_input)
ano_map = anomaly_map[i].squeeze(0).cpu().detach().numpy()
plt.imsave(os.path.join(save_root, class_name, category, fr'{idx_name}_1.png'), ano_map, cmap='jet')
gt_map = gt[i].squeeze(0).cpu().detach().numpy()
plt.imsave(os.path.join(save_root, class_name, category, fr'{idx_name}_2.png'), gt_map, cmap='gray')
plt.close()

**Comment by jun614** [comment]
> Thank you for your interest. We provide a function for saving anomaly maps, which you can use for visualization. This function can be integrated into the evaluation_batch function in utils.py for use. def save_img(imgs, anomaly_map, gt, save_root, img_path): batch_num = imgs.shape[0] for i in range(batch_num): img_path_list = img_path[i].split('') class_name, category, idx_name = img_path_list[-4], img_path_list[-2], img_path_list[-1] os.makedirs(os.path.join(save_root, class_name, category), exist_ok=True) input_frame = denormalize(imgs[i].clone().squeeze(0).cpu().detach().numpy()) cv2_input = np.array(input_frame, dtype=np.uint8) plt.imsave(os.path.join(save_root, class_name, category, fr'{idx_name}_0.png'), cv2_input) ano_map = anomaly_map[i].squeeze(0).cpu().detach().numpy() plt.imsave(os.path.join(save_root, class_name, category, fr'{idx_name}_1.png'), ano_map, cmap='jet') gt_map = gt[i].squeeze(0).cpu().detach().numpy() plt.imsave(os.path.join(save_root, class_name, category, fr'{idx_name}_2.png'), gt_map, cmap='gray') plt.close()

Thank you！
